### ISSUE TITLE
persistent-red-black-node-size: contract violation

### ISSUE THREAD

**Issue by sorawee** [jackfirth/rebellion#510]
```
#lang racket
(require rebellion/base/comparator
         rebellion/collection/sorted-set)

(define *set* (sorted-set #:comparator real<=>))

(sorted-set-element-at-least
 (sorted-set-add *set* 47)
 63)
```

results in:

```
persistent-red-black-node-size: contract violation
  expected: persistent-red-black-node?
  given: 'black-leaf
```

**Comment by jackfirth** [comment]
This bug is fixed in #508 as far as I can tell.

**Comment by jackfirth** [comment]
Closed by #508.
